src/node/desktop/src/main/session-launcher.ts

```diff
@@ -586,6 +587,10 @@ export class SessionLauncher {
       }
     }
 
+    if (process.platform === 'darwin' && process.arch === 'arm64') {
+      checkForRosetta();
```

Comment:
Can we add an escape hatch just in case users need to turn off our Rosetta check for some reason? Perhaps via a desktop option, or an environment variable?

Reply:
I've added an ElectronDesktopOption `checkForRosetta` which is set to true by default.

If the user clicks the `Don't Remind Me Again` button, it will set that option to false.